util/types/field_type.go

```diff
@@ -346,6 +347,9 @@ func DefaultTypeForValue(value interface{}, tp *FieldType) {
 		SetBinChsClnFlag(tp)
 	case json.JSON:
 		tp.Tp = mysql.TypeJSON
+		tp.Decimal = UnspecifiedLength
```

Comment:
1. why set `Decimal` with `UnspecifiedLength ` ?
in mysql:
```sql
create table t(a json);
MySQL > select a from t;
Field   1:  `a`
Catalog:    `def`
Database:   `test`
Table:      `t`
Org_table:  `t`
Type:       JSON
Collation:  binary (63)
Length:     4294967295
Max_length: 0
Decimals:   0
Flags:      BLOB BINARY
```
2. should we set `Flen`  here ?